include/grpc++/impl/codegen/async_stream.h

```diff
@@ -431,6 +514,42 @@ class ServerAsyncWriter final : public ServerAsyncWriterInterface<W> {
     call_.PerformOps(&write_ops_);
   }
 
+  void Write(const W& msg, WriteOptions options, void* tag) override {
```

Comment:
There's a lot of code repetition between Write, WriteAndFinish, and Finish, and that should be turned into code reuse for good code health; otherwise it'll be too easy to slip up and make a change in one place and not the others.